src/backend/commands/dirtablecmds.c

```diff
@@ -135,7 +135,24 @@ CreateDirectoryTable(CreateDirectoryTableStmt *stmt, Oid relId)
 	relFileNode.spcNode = spcId;
 	relFileNode.dbNode = MyDatabaseId;
 	relFileNode.relNode = pg_class_tuple->relfilenode;
-	dirTablePath = UFileFormatPathName(&relFileNode);
+	if (stmt->location)
```

Comment:
What's the valid pattern of location? If the location name is a number, will it conflict with existing or future path? See function `GetDatabasePath()`.

Reply:
There is no pattern specification for location which is not easy to define, if the path has existed, will error out.

Reply:
The integer is used to represent a folder for database by the oid. We'd better to avoid this conflict, i.e. disallowed to use a number as location name.

Reply:
In my opinion, it's maybe better to not do this, which should be defined a specification known by the user and will make the system more complicated. What's more, in some cases, we should replace the location with old location which include integers or oids.